### ISSUE TITLE
Rule proposal: `prefer-map`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1285]
If an object is created from pairs, and only used to check existence and READ properties. Should prefer use `new Map()` instead of `Object.fromEntries()`

## Fail

```js
const object = Object.fromEntries(/*...*/);

if (object.foo) {
	console.log(object.foo)
}
```

## Pass

```js
const object = new Map(/*...*/);

if (object.has('foo')) {
	console.log(object.get('foo'))
}
```

```js
// Object literal should be ignored
const object = {
// ...
};

if (object.foo) {
	console.log(object.foo)
}
```

**Comment by fisker** [comment]
I think this part "and only used to check existence and READ properties. " need a update.

Allow, following use

- `object.foo` -> `map.get('foo')`, we can add another rule `prefer-map-has` to fix it to `map.has` if necessary.
- `Reflect.has(object, foo)` -> `map.has(foo)`
- `delete object.foo` -> `map.delete('foo')`
- `object.foo = bar` -> `map.set('foo', bar)`
- `Object.keys(object)` -> `[...map.keys()]` or `map.keys()`, `Object.entries(object)`, `Object.values(object)` the same.
- More?


Summary, if an object created from `Object.fromEntries()` and only used in cases above, auto replace `Object.fromEntries` to `new Map`

**Comment by sindresorhus** [comment]
This is now accepted.
